data-prepper-test-common/src/test/java/org/opensearch/dataprepper/matcher/MapEqualsTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.matcher;
+
+import org.junit.jupiter.api.Test;
+
+import java.time.OffsetDateTime;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+class MapEqualsTest {
```

Comment:
These are good tests, but is there a way to test the `isEqualWithoutTimestamp` assertion itself? It is some assertion inception though so it may not be as simple as that.

Something like `assertThat(assertThat(actualMap, isEqualWithoutTimestamp(expectedMap)), "passes")`. It may not work like this at all though.

Reply:
I don't think we can test it. But testing `matchesSafely` will ensure logic is correct.

Reply:
Why not just?

```
assertThat(actualMap, isEqualWithoutTimestamp(knownEqualMap));
assertThat(actualMap, not(isEqualWithoutTimestamp(knownDifferentMap)));
```

Reply:
Genius haha